bookkeeper-server/src/main/java/org/apache/bookkeeper/server/EmbeddedServer.java

```diff
@@ -0,0 +1,648 @@
+                           RegistrationManager registrationManager, LedgerManagerFactory ledgerManagerFactory,
+                           DiskChecker diskChecker, LedgerDirsManager ledgerDirsManager,
+                           LedgerDirsManager indexDirsManager, BookieService bookieService,
+                           AutoRecoveryService autoRecoveryService, DataIntegrityService dataIntegrityService,
+                           HttpService httpService) {
+        this.lifecycleComponentStack = lifecycleComponentStack;
+        this.statsProvider = statsProvider;
+        this.registrationManager = registrationManager;
+        this.ledgerManagerFactory = ledgerManagerFactory;
+        this.diskChecker = diskChecker;
+        this.ledgerDirsManager = ledgerDirsManager;
+        this.indexDirsManager = indexDirsManager;
+        this.bookieService = bookieService;
+        this.autoRecoveryService = autoRecoveryService;
+        this.dataIntegrityService = dataIntegrityService;
+        this.httpService = httpService;
+    }
+
+    public LifecycleComponentStack getLifecycleComponentStack() {
+        return lifecycleComponentStack;
+    }
+
+    public StatsProvider getStatsProvider() {
+        return statsProvider;
+    }
+
+    public RegistrationManager getRegistrationManager() {
+        return registrationManager;
+    }
+
+    public LedgerManagerFactory getLedgerManagerFactory() {
+        return ledgerManagerFactory;
+    }
+
+    public DiskChecker getDiskChecker() {
+        return diskChecker;
+    }
+
+    public LedgerDirsManager getLedgerDirsManager() {
+        return ledgerDirsManager;
+    }
+
+    public LedgerDirsManager getIndexDirsManager() {
+        return indexDirsManager;
+    }
+
+    public BookieService getBookieService() {
+        return bookieService;
+    }
+
+    public AutoRecoveryService getAutoRecoveryService() {
+        return autoRecoveryService;
+    }
+
+    public DataIntegrityService getDataIntegrityService() {
+        return dataIntegrityService;
+    }
+
+    public HttpService getHttpService() {
+        return httpService;
+    }
+
+    /**
+     * Create a new builder from given configuration. Actual services implementations can be provided to the builder and
+     * will override ones defined in the configuration.
+     * <p>
+     * Invoker is responsible to start and stop provided services implementations, components from
+     * {@link EmbeddedServer#getLifecycleComponentStack()} will reflect only those created from provided configuration.
+     *
+     * @param conf bookie configuration
+     * @return a new embedded server builder
+     */
+    public static final Builder builder(BookieConfiguration conf) {
+        return new Builder(conf);
+    }
+
+    @Slf4j
+    public static class Builder {
+
+        private BookieConfiguration conf;
+
+        boolean addExternalResourcesToLifecycle = true;
```

Comment:
What does this mean?
If a component is passed from outside we shouldn't handle it.
Why are you adding this feature?
Also, if you really need this it must be turned off by default

Reply:
Sorry is a WIP fix but I had to see running it on CI

Reply:
Set to default false in the last commit

Reply:
if this is not required to make CI pass I prefer to drop the flag at all.
Otherwise we add a feature that won't be used by anyone but we will have to support it forever.
Currently in the case of Pravega, Pulsar and HerdDB that I know, there is no need to let the Bookie take control of externally provided components

Reply:
Removed addExternalResourcesToLifecycle